tests/unittests/Makefile

```diff
@@ -61,3 +61,9 @@ ifeq (, $(UNIT_TESTS))
 else
   CFLAGS += -DTEST_SUITES='$(subst $() $(),$(comma),$(UNIT_TESTS:tests-%=%))'
 endif
+
+# Hack: If GNRC is not used, still provide access to sock_types.h to allow
+# building nanocoap
+ifeq (,$(filter gnrc_sock,$(USEMODULE)))
+  CFLAGS +=-I$(RIOTBASE)/sys/net/gnrc/sock/include
+endif
```

Comment:
I like the hack of providing a mock instead of the full header better. :sweat_smile: Why can't it be just moved to before `tests-nanocoap*` include `net/nanocoap.h`?

Reply:
That would not work. Other C files also include nanocoap.h. And also, it is not just the type but also the function checking for multicast endpoints that is needed.

Reply:
_grumble_ fine...

Reply:
I pinky promise to separate the CoAP packet building and parsing from the network stuff when I find time for that. When this is in, we don't need that hack aaymore :)